// @jgoz/esbuild-plugin-livereload/dist/overlay-XHYH2LTV.js
// ../esbuild-overlay/dist/overlay.js
function F() {
}
function at(t2) {
  return t2();
}
function bt() {
  return /* @__PURE__ */ Object.create(null);
}
function U(t2) {
  t2.forEach(at);
}
function q(t2) {
  return typeof t2 == "function";
}
function I(t2, e) {
  return t2 != t2 ? e == e : t2 !== e || t2 && typeof t2 == "object" || typeof t2 == "function";
}
function kt(t2) {
  return Object.keys(t2).length === 0;
}
var Ft = typeof window < "u" ? window : typeof globalThis < "u" ? globalThis : global;
var lt = class t {
  constructor(e) {
    this.options = e, this._listeners = "WeakMap" in Ft ? /* @__PURE__ */ new WeakMap() : void 0;
  }
  observe(e, n) {
    return this._listeners.set(e, n), this._getObserver().observe(e, this.options), () => {
      this._listeners.delete(e), this._observer.unobserve(e);
    };
  }
  _getObserver() {
    var e;
    return (e = this._observer) !== null && e !== void 0 ? e : this._observer = new ResizeObserver((n) => {
      var i;
      for (let o of n) t.entries.set(o.target, o), (i = this._listeners.get(o.target)) === null || i === void 0 || i(o);
    });
  }
};
lt.entries = "WeakMap" in Ft ? /* @__PURE__ */ new WeakMap() : void 0;
var xt = false;
function Nt() {
  xt = true;
}
function Rt() {
  xt = false;
}
function c(t2, e) {
  t2.appendChild(e);
}
function B(t2, e, n) {
  let i = jt(t2);
  if (!i.getElementById(e)) {
    let o = p("style");
    o.id = e, o.textContent = n, zt(i, o);
  }
}
function jt(t2) {
  if (!t2) return document;
  let e = t2.getRootNode ? t2.getRootNode() : t2.ownerDocument;
  return e && e.host ? e : t2.ownerDocument;
}
function zt(t2, e) {
  return c(t2.head || t2, e), e.sheet;
}
function M(t2, e, n) {
  t2.insertBefore(e, n || null);
}
function O(t2) {
  t2.parentNode && t2.parentNode.removeChild(t2);
}
function K(t2, e) {
  for (let n = 0; n < t2.length; n += 1) t2[n] && t2[n].d(e);
}
function p(t2) {
  return document.createElement(t2);
}
function x(t2) {
  return document.createTextNode(t2);
}
function C() {
  return x(" ");
}
function Q(t2, e, n, i) {
  return t2.addEventListener(e, n, i), () => t2.removeEventListener(e, n, i);
}
function g(t2, e, n) {
  n == null ? t2.removeAttribute(e) : t2.getAttribute(e) !== n && t2.setAttribute(e, n);
}
function Pt(t2) {
  return Array.from(t2.childNodes);
}
function Z(t2, e) {
  e = "" + e, t2.data !== e && (t2.data = e);
}
var ft;
function X(t2) {
  ft = t2;
}
var z = [];
var yt = [];
var P = [];
var wt = [];
var qt = Promise.resolve();
var ct = false;
function Ut() {
  ct || (ct = true, qt.then($t));
}
function ut(t2) {
  P.push(t2);
}
var st = /* @__PURE__ */ new Set();
var j = 0;
function $t() {
  if (j !== 0) return;
  let t2 = ft;
  do {
    try {
      for (; j < z.length; ) {
        let e = z[j];
        j++, X(e), It(e.$$);
      }
    } catch (e) {
      throw z.length = 0, j = 0, e;
    }
    for (X(null), z.length = 0, j = 0; yt.length; ) yt.pop()();
    for (let e = 0; e < P.length; e += 1) {
      let n = P[e];
      st.has(n) || (st.add(n), n());
    }
    P.length = 0;
  } while (z.length);
  for (; wt.length; ) wt.pop()();
  ct = false, st.clear(), X(t2);
}
function It(t2) {
  if (t2.fragment !== null) {
    t2.update(), U(t2.before_update);
    let e = t2.dirty;
    t2.dirty = [-1], t2.fragment && t2.fragment.p(t2.ctx, e), t2.after_update.forEach(ut);
  }
}
function Bt(t2) {
  let e = [], n = [];
  P.forEach((i) => t2.indexOf(i) === -1 ? e.push(i) : n.push(i)), n.forEach((i) => i()), P = e;
}
var J = /* @__PURE__ */ new Set();
var R;
function tt() {
  R = { r: 0, c: [], p: R };
}
function et() {
  R.r || U(R.c), R = R.p;
}
function E(t2, e) {
  t2 && t2.i && (J.delete(t2), t2.i(e));
}
function A(t2, e, n, i) {
  if (t2 && t2.o) {
    if (J.has(t2)) return;
    J.add(t2), R.c.push(() => {
      J.delete(t2), i && (n && t2.d(1), i());
    }), t2.o(e);
  } else i && i();
}
var Wt = ["allowfullscreen", "allowpaymentrequest", "async", "autofocus", "autoplay", "checked", "controls", "default", "defer", "disabled", "formnovalidate", "hidden", "inert", "ismap", "loop", "multiple", "muted", "nomodule", "novalidate", "open", "playsinline", "readonly", "required", "reversed", "selected"];
var re = /* @__PURE__ */ new Set([...Wt]);
function nt(t2) {
  t2 && t2.c();
}
function Y(t2, e, n, i) {
  let { fragment: o, after_update: l } = t2.$$;
  o && o.m(e, n), i || ut(() => {
    let f = t2.$$.on_mount.map(at).filter(q);
    t2.$$.on_destroy ? t2.$$.on_destroy.push(...f) : U(f), t2.$$.on_mount = [];
  }), l.forEach(ut);
}
function W(t2, e) {
  let n = t2.$$;
  n.fragment !== null && (Bt(n.after_update), U(n.on_destroy), n.fragment && n.fragment.d(e), n.on_destroy = n.fragment = null, n.ctx = []);
}
function Ht(t2, e) {
  t2.$$.dirty[0] === -1 && (z.push(t2), Ut(), t2.$$.dirty.fill(0)), t2.$$.dirty[e / 31 | 0] |= 1 << e % 31;
}
function H(t2, e, n, i, o, l, f, a = [-1]) {
  let h = ft;
  X(t2);
  let r = t2.$$ = { fragment: null, ctx: [], props: l, update: F, not_equal: o, bound: bt(), on_mount: [], on_destroy: [], on_disconnect: [], before_update: [], after_update: [], context: new Map(e.context || (h ? h.$$.context : [])), callbacks: bt(), dirty: a, skip_bound: false, root: e.target || h.$$.root };
  f && f(r.root);
  let $ = false;
  if (r.ctx = n ? n(t2, e.props || {}, (u, s, ...S) => {
    let d = S.length ? S[0] : s;
    return r.ctx && o(r.ctx[u], r.ctx[u] = d) && (!r.skip_bound && r.bound[u] && r.bound[u](d), $ && Ht(t2, u)), s;
  }) : [], r.update(), $ = true, U(r.before_update), r.fragment = i ? i(r.ctx) : false, e.target) {
    if (e.hydrate) {
      Nt();
      let u = Pt(e.target);
      r.fragment && r.fragment.l(u), u.forEach(O);
    } else r.fragment && r.fragment.c();
    e.intro && E(t2.$$.fragment), Y(t2, e.target, e.anchor, e.customElement), Rt(), $t();
  }
  X(h);
}
var Gt;
typeof HTMLElement == "function" && (Gt = class extends HTMLElement {
  constructor() {
    super(), this.attachShadow({ mode: "open" });
  }
  connectedCallback() {
    let { on_mount: t2 } = this.$$;
    this.$$.on_disconnect = t2.map(at).filter(q);
    for (let e in this.$$.slotted) this.appendChild(this.$$.slotted[e]);
  }
  attributeChangedCallback(t2, e, n) {
    this[t2] = n;
  }
  disconnectedCallback() {
    U(this.$$.on_disconnect);
  }
  $destroy() {
    W(this, 1), this.$destroy = F;
  }
  $on(t2, e) {
    if (!q(e)) return F;
    let n = this.$$.callbacks[t2] || (this.$$.callbacks[t2] = []);
    return n.push(e), () => {
      let i = n.indexOf(e);
      i !== -1 && n.splice(i, 1);
    };
  }
  $set(t2) {
    this.$$set && !kt(t2) && (this.$$.skip_bound = true, this.$$set(t2), this.$$.skip_bound = false);
  }
});
var D = class {
  $destroy() {
    W(this, 1), this.$destroy = F;
  }
  $on(e, n) {
    if (!q(n)) return F;
    let i = this.$$.callbacks[e] || (this.$$.callbacks[e] = []);
    return i.push(n), () => {
      let o = i.indexOf(n);
      o !== -1 && i.splice(o, 1);
    };
  }
  $set(e) {
    this.$$set && !kt(e) && (this.$$.skip_bound = true, this.$$set(e), this.$$.skip_bound = false);
  }
};
function Xt(t2) {
  B(t2, "svelte-1ua6zs9", ".line-text.svelte-1ua6zs9 strong.svelte-1ua6zs9{background:#ffff78;color:#5f4545}");
}
function Yt(t2) {
  let e, n, i, o;
  return { c() {
    e = p("span"), n = x(t2[0]), i = p("strong"), i.textContent = `${t2[1]}`, o = x(t2[2]), g(i, "class", "svelte-1ua6zs9"), g(e, "class", "line-text svelte-1ua6zs9");
  }, m(l, f) {
    M(l, e, f), c(e, n), c(e, i), c(e, o);
  }, p: F, i: F, o: F, d(l) {
    l && O(e);
  } };
}
function Vt(t2, e, n) {
  let { text: i } = e, { column: o } = e, { length: l } = e, [f, a, h] = [i.slice(0, o), i.slice(o, o + l), i.slice(o + l)];
  return t2.$$set = (r) => {
    "text" in r && n(3, i = r.text), "column" in r && n(4, o = r.column), "length" in r && n(5, l = r.length);
  }, [f, a, h, i, o, l];
}
var dt = class extends D {
  constructor(e) {
    super(), H(this, e, Vt, Yt, I, { text: 3, column: 4, length: 5 }, Xt);
  }
};
var Et = dt;
function Jt(t2) {
  B(t2, "svelte-1ppwwbh", ".stack-entry.svelte-1ppwwbh.svelte-1ppwwbh{cursor:pointer;margin-bottom:2.5em}.lines.svelte-1ppwwbh.svelte-1ppwwbh{color:rgb(187, 165, 165);font-size:0.835em}.lines.svelte-1ppwwbh.svelte-1ppwwbh:not(.no-fade){-webkit-mask-image:linear-gradient(to bottom, rgba(0, 0, 0, 1) 75%, rgba(0, 0, 0, 0));mask-image:linear-gradient(to bottom, rgba(0, 0, 0, 1) 75%, rgba(0, 0, 0, 0))}.line-number.svelte-1ppwwbh.svelte-1ppwwbh{padding-right:1.5em;opacity:0.5}.stack-entry.svelte-1ppwwbh:first-child .line-hili strong.svelte-1ppwwbh{text-decoration:underline wavy #ff0040}.file.svelte-1ppwwbh.svelte-1ppwwbh{font-weight:bold;margin-top:2.5em;margin-bottom:1.5em;color:rgb(202, 17, 63)}.file.svelte-1ppwwbh strong.svelte-1ppwwbh{text-decoration:underline}.file.svelte-1ppwwbh.svelte-1ppwwbh:before{content:'@ ';opacity:0.5;margin-left:-1.25em}");
}
function St(t2, e, n) {
  let i = t2.slice();
  return i[15] = e[n][0], i[8] = e[n][1], i;
}
function Kt(t2) {
  let e, n = String(t2[15]).padStart(t2[5], "  ") + "", i;
  return { c() {
    e = p("span"), i = x(n), g(e, "class", "line-number svelte-1ppwwbh");
  }, m(o, l) {
    M(o, e, l), c(e, i);
  }, p: F, d(o) {
    o && O(e);
  } };
}
function Qt(t2) {
  let e = t2[8] + "", n;
  return { c() {
    n = x(e);
  }, m(i, o) {
    M(i, n, o);
  }, p: F, i: F, o: F, d(i) {
    i && O(n);
  } };
}
function Zt(t2) {
  let e, n;
  return e = new Et({ props: { text: t2[8], column: t2[1], length: t2[3] } }), { c() {
    nt(e.$$.fragment);
  }, m(i, o) {
    Y(e, i, o), n = true;
  }, p: F, i(i) {
    n || (E(e.$$.fragment, i), n = true);
  }, o(i) {
    A(e.$$.fragment, i), n = false;
  }, d(i) {
    W(e, i);
  } };
}
function Ot(t2) {
  let e, n, i, o, l, f, a = t2[15] > 0 && Kt(t2), h = [Zt, Qt], r = [];
  function $(u, s) {
    return u[15] == u[2] ? 0 : 1;
  }
  return i = $(t2, -1), o = r[i] = h[i](t2), { c() {
    e = p("div"), a && a.c(), n = C(), o.c(), l = C(), g(e, "class", "line");
  }, m(u, s) {
    M(u, e, s), a && a.m(e, null), c(e, n), r[i].m(e, null), c(e, l), f = true;
  }, p(u, s) {
    u[15] > 0 && a.p(u, s), o.p(u, s);
  }, i(u) {
    f || (E(o), f = true);
  }, o(u) {
    A(o), f = false;
  }, d(u) {
    u && O(e), a && a.d(), r[i].d();
  } };
}
function te(t2) {
  let e, n, i, o, l, f, a, h, r, $, u = t2[6], s = [];
  for (let d = 0; d < u.length; d += 1) s[d] = Ot(St(t2, u, d));
  let S = (d) => A(s[d], 1, 1, () => {
    s[d] = null;
  });
  return { c() {
    e = p("div"), n = p("div"), i = p("strong"), i.textContent = `${t2[4]}`, o = C(), l = p("div"), l.textContent = `${t2[0]}`, f = C(), a = p("div");
    for (let d = 0; d < s.length; d += 1) s[d].c();
    g(i, "class", "svelte-1ppwwbh"), g(n, "class", "file svelte-1ppwwbh"), g(l, "class", "error"), g(a, "class", "lines svelte-1ppwwbh"), g(e, "class", "stack-entry svelte-1ppwwbh");
  }, m(d, y) {
    M(d, e, y), c(e, n), c(n, i), c(e, o), c(e, l), c(e, f), c(e, a);
    for (let _ = 0; _ < s.length; _ += 1) s[_] && s[_].m(a, null);
    h = true, r || ($ = Q(e, "click", t2[7]), r = true);
  }, p(d, [y]) {
    if (y & 110) {
      u = d[6];
      let _;
      for (_ = 0; _ < u.length; _ += 1) {
        let T = St(d, u, _);
        s[_] ? (s[_].p(T, y), E(s[_], 1)) : (s[_] = Ot(T), s[_].c(), E(s[_], 1), s[_].m(a, null));
      }
      for (tt(), _ = u.length; _ < s.length; _ += 1) S(_);
      et();
    }
  }, i(d) {
    if (!h) {
      for (let y = 0; y < u.length; y += 1) E(s[y]);
      h = true;
    }
  }, o(d) {
    s = s.filter(Boolean);
    for (let y = 0; y < s.length; y += 1) A(s[y]);
    h = false;
  }, d(d) {
    d && O(e), K(s, d), r = false, $();
  } };
}
function ee(t2, e, n) {
  var i;
  let o = { length: 0, line: 1, lineText: "???", column: 1, file: "(no file)", namespace: "", suggestion: "" }, { error: l } = e, { openFileURL: f } = e, a = (i = l.location) !== null && i !== void 0 ? i : o, { text: h } = l, { column: r, line: $, length: u, lineText: s, file: S } = a, d = String($).length, y = s.split(/\r?\n/g), _ = y.length > 1 ? y.map((w) => [0, w]) : [[$, s]];
  function T() {
    if (!f) return;
    let w = new URL(f);
    w.searchParams.append("file", a.file), w.searchParams.append("line", String(a.line)), w.searchParams.append("column", String(a.column)), fetch(w.toString());
  }
  return t2.$$set = (w) => {
    "error" in w && n(9, l = w.error), "openFileURL" in w && n(10, f = w.openFileURL);
  }, [h, r, $, u, S, d, _, T, s, l, f];
}
var _t = class extends D {
  constructor(e) {
    super(), H(this, e, ee, te, I, { error: 9, openFileURL: 10 }, Jt);
  }
};
var Ct = _t;
function ne(t2) {
  B(t2, "svelte-1nkg4nv", ".modal.svelte-1nkg4nv.svelte-1nkg4nv{position:fixed;top:0;right:0;bottom:0;left:0;background:white;z-index:10000;box-sizing:border-box;overflow-y:scroll;overflow-x:hidden;font-size:18px;--left-pad:60px}.modal.svelte-1nkg4nv.svelte-1nkg4nv,.modal.svelte-1nkg4nv .svelte-1nkg4nv{display:block;padding:0;margin:0;font-family:Menlo, Monaco, 'Courier New', Courier, monospace}.modal.svelte-1nkg4nv h1.svelte-1nkg4nv{color:black;margin:0;padding:0;font-size:1.77em;font-weight:600;opacity:0.75;margin-top:50px;margin-bottom:45px;position:relative;padding-left:var(--left-pad)}.close.svelte-1nkg4nv.svelte-1nkg4nv{all:unset;color:black;font-weight:normal;text-decoration:none;position:absolute;top:-0.32em;right:1em;font-size:1.77em;opacity:0.15;transition:all 0.25s ease-in-out}.close.svelte-1nkg4nv.svelte-1nkg4nv:hover{transform:scale(1.5);opacity:0.25}.error.svelte-1nkg4nv.svelte-1nkg4nv{margin:1em 0 3em 0;left:0}.error-title.svelte-1nkg4nv.svelte-1nkg4nv{display:flex;align-items:stretch;padding-right:50px}.error-type.svelte-1nkg4nv.svelte-1nkg4nv{min-height:2.8em;display:flex !important;align-items:center;padding:0 1em;background:rgb(255, 0, 64);color:white;margin-right:2em;padding-left:var(--left-pad);white-space:nowrap}.error-counter.svelte-1nkg4nv.svelte-1nkg4nv{color:white;opacity:0.3;position:absolute;left:0.8em}.error-message.svelte-1nkg4nv.svelte-1nkg4nv{display:flex !important;align-items:center;font-weight:400;line-height:1em}.error-stack.svelte-1nkg4nv.svelte-1nkg4nv{margin-top:2em;white-space:pre;padding-left:var(--left-pad)}@media only screen and (max-width: 640px){.modal.svelte-1nkg4nv.svelte-1nkg4nv{font-size:15px;--left-pad:50px}.modal.svelte-1nkg4nv h1.svelte-1nkg4nv{margin:40px 0}}@media only screen and (max-width: 500px){.modal.svelte-1nkg4nv.svelte-1nkg4nv{font-size:14px;--left-pad:45px}.modal.svelte-1nkg4nv h1.svelte-1nkg4nv{margin:30px 0}}");
}
function Mt(t2, e, n) {
  let i = t2.slice();
  return i[5] = e[n], i;
}
function At(t2) {
  let e, n;
  return e = new Ct({ props: { error: t2[5], openFileURL: t2[1] } }), { c() {
    nt(e.$$.fragment);
  }, m(i, o) {
    Y(e, i, o), n = true;
  }, p(i, o) {
    let l = {};
    o & 1 && (l.error = i[5]), o & 2 && (l.openFileURL = i[1]), e.$set(l);
  }, i(i) {
    n || (E(e.$$.fragment, i), n = true);
  }, o(i) {
    A(e.$$.fragment, i), n = false;
  }, d(i) {
    W(e, i);
  } };
}
function ie(t2) {
  let e, n, i, o, l, f, a, h, r, $, u, s, S = t2[0].length + "", d, y, _, T, w = t2[0].length + "", it, pt, V = t2[0].length > 1 ? "s" : "", ot, mt, G, L, rt, gt, N = t2[0], m = [];
  for (let v = 0; v < N.length; v += 1) m[v] = At(Mt(t2, N, v));
  let Lt = (v) => A(m[v], 1, 1, () => {
    m[v] = null;
  });
  return { c() {
    e = p("div"), n = p("h1"), i = x("Oops :( "), o = p("button"), o.textContent = "\xD7", l = C(), f = p("div"), a = p("div"), h = p("div"), r = p("span"), $ = x(t2[3]), u = C(), s = p("span"), d = x(S), y = C(), _ = p("span"), T = x("Build failed with "), it = x(w), pt = x(" error"), ot = x(V), mt = C(), G = p("div");
    for (let v = 0; v < m.length; v += 1) m[v].c();
    g(o, "class", "close svelte-1nkg4nv"), g(n, "class", "svelte-1nkg4nv"), g(s, "class", "error-counter svelte-1nkg4nv"), g(r, "class", "error-type svelte-1nkg4nv"), g(_, "class", "error-message svelte-1nkg4nv"), g(h, "class", "error-title svelte-1nkg4nv"), g(G, "class", "error-stack svelte-1nkg4nv"), g(a, "class", "error svelte-1nkg4nv"), g(f, "class", "errors svelte-1nkg4nv"), g(e, "class", "modal svelte-1nkg4nv");
  }, m(v, k) {
    M(v, e, k), c(e, n), c(n, i), c(n, o), c(e, l), c(e, f), c(f, a), c(a, h), c(h, r), c(r, $), c(r, u), c(r, s), c(s, d), c(h, y), c(h, _), c(_, T), c(_, it), c(_, pt), c(_, ot), c(a, mt), c(a, G);
    for (let b = 0; b < m.length; b += 1) m[b] && m[b].m(G, null);
    L = true, rt || (gt = Q(o, "click", function() {
      q(t2[2]) && t2[2].apply(this, arguments);
    }), rt = true);
  }, p(v, [k]) {
    if (t2 = v, (!L || k & 1) && S !== (S = t2[0].length + "") && Z(d, S), (!L || k & 1) && w !== (w = t2[0].length + "") && Z(it, w), (!L || k & 1) && V !== (V = t2[0].length > 1 ? "s" : "") && Z(ot, V), k & 3) {
      N = t2[0];
      let b;
      for (b = 0; b < N.length; b += 1) {
        let vt = Mt(t2, N, b);
        m[b] ? (m[b].p(vt, k), E(m[b], 1)) : (m[b] = At(vt), m[b].c(), E(m[b], 1), m[b].m(G, null));
      }
      for (tt(), b = N.length; b < m.length; b += 1) Lt(b);
      et();
    }
  }, i(v) {
    if (!L) {
      for (let k = 0; k < N.length; k += 1) E(m[k]);
      L = true;
    }
  }, o(v) {
    m = m.filter(Boolean);
    for (let k = 0; k < m.length; k += 1) A(m[k]);
    L = false;
  }, d(v) {
    v && O(e), K(m, v), rt = false, gt();
  } };
}
function oe(t2, e, n) {
  var i;
  let { errors: o } = e, { openFileURL: l } = e, { onClose: f } = e, a = o.length > 0 && ((i = o[0].detail) === null || i === void 0 ? void 0 : i.type) || "Error";
  return t2.$$set = (h) => {
    "errors" in h && n(0, o = h.errors), "openFileURL" in h && n(1, l = h.openFileURL), "onClose" in h && n(2, f = h.onClose);
  }, [o, l, f, a];
}
var ht = class extends D {
  constructor(e) {
    super(), H(this, e, oe, ie, I, { errors: 0, openFileURL: 1, onClose: 2 }, ne);
  }
};
var Dt = ht;
var Tt = "__esbuild_lr_root__";
function ge(t2) {
  let e = document.getElementById(Tt);
  e || (e = document.createElement("div"), e.setAttribute("id", Tt), document.body.appendChild(e));
  let n = () => {
    i.$destroy(), e && e.remove();
  }, i = new Dt({ target: e, props: { errors: t2.errors, openFileURL: t2.openFileURL, onClose: n } });
  return n;
}
export {
  ge as overlay
};
